torchgeo/datasets/cropharvest.py

```diff
@@ -0,0 +1,317 @@
+            its key for associated data
+        """
+        files = []
+        chips = glob.glob(
+            os.path.join(
+                root,
+                directory,
+                self.file_dict["features"]["extracted_filename"],
+                "*.h5",
+            )
+        )
+        chips = sorted(os.path.basename(chip) for chip in chips)
+        for chip in chips:
+            chip_path = os.path.join(
+                root, directory, self.file_dict["features"]["extracted_filename"], chip
+            )
+            index = chip.split("_")[0]
+            dataset = chip.split("_")[1][:-3]
+            files.append(dict(chip=chip_path, index=index, dataset=dataset))
+        return files
+
+    def _load_labels(self, root: str, directory: str) -> pd.DataFrame:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+            directory: sub directory CropHarvest
+
+        Returns:
+            pandas dataframe containing label data for each feature
+        """
+        filename = self.file_dict["labels"]["extracted_filename"]
+        with open(os.path.join(root, directory, filename), encoding="utf8") as f:
+            data = json.load(f)
+
+            pd.json_normalize(data["features"])
+            df = pd.json_normalize(data["features"])
+
+            return df
+
+    def _load_array(self, path: str) -> Tensor:
+        """Load an individual single pixel timeseries.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+            ImportError if h5py is not installed
+        """
+        try:
+            import h5py  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "h5py is not installed and is required to use this dataset"
+            )
+        filename = os.path.join(path)
+        with h5py.File(filename, "r") as f:
+            array = f.get("array")[()]
+            tensor = torch.from_numpy(array).float()
+            return tensor
+
+    def _load_label(self, idx: str, dataset: str) -> str:
+        """Load the croptype label for a single pixel timeseries.
+
+        Args:
+            idx: sample index in labels.geojson
+            dataset: dataset name to query labels.geojson
+        Returns:
+            the croptype label,
+            "Some" if no croptype is defined but is_crop is true,
+            or "None" if there are no crops
+        """
+        index = int(idx)
+        row = self.labels[
+            (self.labels["properties.index"] == index)
+            & (self.labels["properties.dataset"] == dataset)
+        ]
+        row = row.to_dict(orient="records")[0]
+        label = "None"
+        if row["properties.label"]:
+            label = row["properties.label"]
+        elif row["properties.is_crop"] == 1:
+            label = "Some"
+
+        return label
+
+    def _check_integrity(self) -> bool:
```

Comment:
This follows the old dataset style, I would instead base this off the newer style. Search similar datasets for `_verify`.